project3/common/move/move.c

```diff
@@ -0,0 +1,72 @@
+#include "move.h"
+
+void stop(Move *move) {
+    move->left_speed = 0;
+    move->right_speed = 0;
+}
+
+void forward(Move *move, int16_t speed) {
+    move->left_speed = speed;
+    move->right_speed = speed;
+}
+
+void backward(Move *move, int16_t speed) {
+    move->left_speed = -speed;
+    move->right_speed = -speed;
+}
+
+void spin_right(Move *move, int16_t speed) {
+    move->left_speed = speed;
+    move->right_speed = -speed;
+}
+
+void spin_left(Move *move, int16_t speed) {
+    move->left_speed = -speed;
+    move->right_speed = speed;
+}
+
+void set_speeds(Move *move, int16_t left_speed, int16_t right_speed) {
+    move->left_speed = left_speed;
+    move->right_speed = right_speed;
+}
+
+int16_t roomba_speed(int16_t speed) {
+    speed = constrain_u(speed, -100, 100);
+    return speed < 0
```

Comment:
I think this whole function can be replaced with

```c
int16_t roomba_speed(int16_t speed) {
    return cmap_u(speed, -100, 100, -MAX_SPEED, MAX_SPEED);
}
```

Reply:
It can 👍